File: src/APIBundle/Entity/Grant.php
<?php

namespace APIBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity()
 * @ORM\Table(name="grants", uniqueConstraints={@ORM\UniqueConstraint(name="grant", columns={"grant_group", "grant_exercise", "grant_name"})})
 */
class Grant
{
    /**
     * @ORM\Id
     * @ORM\Column(type="string")
     * @ORM\GeneratedValue(strategy="UUID")
     */
    protected $grant_id;

    /**
     * @ORM\ManyToOne(targetEntity="Group", inversedBy="group_grants")
     * @ORM\JoinColumn(name="grant_group", referencedColumnName="group_id", onDelete="CASCADE")
     * @var Group
     */
    protected $grant_group;

    /**
     * @ORM\ManyToOne(targetEntity="Exercise")
     * @ORM\JoinColumn(name="grant_exercise", referencedColumnName="exercise_id", onDelete="CASCADE")
     * @var Exercise
     */
    protected $grant_exercise;

    /**
     * @ORM\Column(type="string", columnDefinition="ENUM('ADMIN', 'PLANNER', 'PLAYER', 'OBSERVER')")
     */
    protected $grant_name;

    public function getGrantId()
    {
        return $this->grant_id;
    }

    public function setGrantId($id)
    {
        $this->grant_id = $id;
        return $this;
    }

    public function getGrantGroup()
    {
        return $this->grant_group;
    }

    public function setGrantGroup($group)
    {
        $this->grant_group = $group;
        return $this;
    }

    public function getGrantExercise()
    {
        return $this->grant_exercise;
    }

    public function setGrantExercise($exercise)
    {
        $this->grant_exercise = $exercise;
        return $this;
    }

    public function getGrantName()
    {
        return $this->grant_name;
    }

    public function setGrantName($name)
    {
        $this->grant_name = $name;
        return $this;
    }
}

File: openex-api/src/main/java/io/openex/database/model/Grant.java
package io.openex.database.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import io.openex.helper.MonoModelDeserializer;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;

@Entity
@Table(name = "grants")
public class Grant implements Base {

    public enum GRANT_TYPE {
        OBSERVER,
        PLANNER
    }

    @Id
    @Column(name = "grant_id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID", strategy = "org.hibernate.id.UUIDGenerator")
    @JsonProperty("grant_id")
    private String id;

    @Column(name = "grant_name")
    @JsonProperty("grant_name")
    private String name;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "grant_group")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("grant_group")
    private Group group;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "grant_exercise")
    @JsonSerialize(using = MonoModelDeserializer.class)
    @JsonProperty("grant_exercise")
    private Exercise exercise;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Group getGroup() {
        return group;
    }

    public void setGroup(Group group) {
        this.group = group;
    }

    public Exercise getExercise() {
        return exercise;
    }

    public void setExercise(Exercise exercise) {
        this.exercise = exercise;
    }
}
